TypeCobol/Analysis/AnalyzerProvider.cs

```diff
@@ -10,103 +12,177 @@ namespace TypeCobol.Analysis
     /// <summary>
     /// IAnalyzerProvider default implementation. Can be used directly by adding activation delegates
     /// with the appropriate AddActivator method or by subclassing it.
+    /// It can also be used to allow composition of multiple providers into one.
+    /// This ensures that the creation of IAnalyzers won't interrupt the parsing.
     /// </summary>
-    public class AnalyzerProvider : IAnalyzerProvider
+    public class AnalyzerProviderWrapper : IAnalyzerProvider
     {
+        /// <summary>
+        /// Method used to log exceptions.
+        /// <remarks>Should not be used directly, use the accessor Logger instead.</remarks>
+        /// </summary>
+        [NotNull]
```

Comment:
The `[NotNull]` attribute should be placed on the protected property but not here.